xc/device.ml

```diff
@@ -1511,7 +1524,10 @@ module Vusb = struct
            Also need to do deprivileged work in usb_reset script if QEMU is deprivileged.
         *)
         begin match Qemu.pid ~xs domid with
-          | Some pid -> usb_reset_attach ~hostbus ~hostport ~domid ~pid ~privileged
+          | Some pid ->
+            let stdout = usb_reset_attach ~hostbus ~hostport ~domid ~pid ~privileged in
```

Comment:
Why are the uid/gid obtained from script output? Would it not be easier to obtain them using `Unix.stat` directly here?

Reply:
Yes. We can do that through `Unix.stat`. The path of the device file is gotten from another file such as follow ing code
```
 let get_num path =
      let in_chan = open_in path in
      let line = input_line in_chan in
      close_in in_chan;
      line
    in
    let busnum_path = Printf.sprintf "/sys/bus/usb/devices/%s-%s/busnum" hostbus hostport in
    let devnum_path = Printf.sprintf "/sys/bus/usb/devices/%s-%s/devnum" hostbus hostport in
    let dev_path = Printf.sprintf "/dev/bus/usb/00%s/00%s" (get_num busnum_path) (get_num devnum_path) in
```